infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/StateTransfer.java

```diff
@@ -219,7 +220,9 @@ private static void transferChunk(Layout layout,
                 log.error("Missing address {} in batch {}", address, chunk);
                 throw new IllegalStateException("Missing address");
             }
-            entries.add((LogData) dataMap.get(address));
+            LogData ld = (LogData) dataMap.get(address);
```

Comment:
If logdata is read from the cluster, how come we should set the payload codec type before writing the entries? Shouldn't this information already be available at this point?

Reply:
I'm not the author, but I'm guessing it's because the state transfer runtime might have been configured top use a different codec, which is a valid use case.